lib/phoenix_live_view/async_result.ex

```diff
@@ -0,0 +1,501 @@
+defmodule Phoenix.LiveView.AsyncResult do
```

Comment:
Nitpick: those functions should be added to Phoenix.Component or Phoenix.LiveView accordingly. If we broke every socket function into modules `P.LV.Stream`, `P.LV.Async`, etc, it would actually get harder and harder to see which functionality is available. In fact, we spent a lot of time unifying it all into either `P.LV` or `P.C` so we don't have to deal with this. :D

The docs can be either moved to the LV module docs, a guide, or to each individial function?